hledger-lib/Hledger/Reports/ReportOptions.hs

```diff
@@ -194,56 +192,51 @@ defreportopts = ReportOpts
     , invert_          = False
     , normalbalance_   = Nothing
     , color_           = False
-    , forecast_        = Nothing
     , transpose_       = False
     }
 
-rawOptsToReportOpts :: RawOpts -> IO ReportOpts
-rawOptsToReportOpts rawopts = do
-    d <- getCurrentDay
-
-    let formatstring = T.pack <$> maybestringopt "format" rawopts
-        querystring  = map T.pack $ listofstringopt "args" rawopts  -- doesn't handle an arg like "" right
-        (costing, valuation) = valuationTypeFromRawOpts rawopts
-
-    format <- case parseStringFormat <$> formatstring of
-        Nothing         -> return defaultBalanceLineFormat
-        Just (Right x)  -> return x
-        Just (Left err) -> fail $ "could not parse format option: " ++ err
-
-    return defreportopts
-          {period_      = periodFromRawOpts d rawopts
-          ,interval_    = intervalFromRawOpts rawopts
-          ,statuses_    = statusesFromRawOpts rawopts
-          ,cost_        = costing
-          ,value_       = valuation
-          ,infer_value_ = boolopt "infer-market-price" rawopts
-          ,depth_       = maybeposintopt "depth" rawopts
-          ,date2_       = boolopt "date2" rawopts
-          ,empty_       = boolopt "empty" rawopts
-          ,no_elide_    = boolopt "no-elide" rawopts
-          ,real_        = boolopt "real" rawopts
-          ,format_      = format
-          ,querystring_ = querystring
-          ,average_     = boolopt "average" rawopts
-          ,related_     = boolopt "related" rawopts
-          ,txn_dates_   = boolopt "txn-dates" rawopts
-          ,balancecalc_ = balancecalcopt rawopts
-          ,balanceaccum_ = balanceaccumopt rawopts
-          ,budgetpat_   = maybebudgetpatternopt rawopts
-          ,accountlistmode_ = accountlistmodeopt rawopts
-          ,drop_        = posintopt "drop" rawopts
-          ,row_total_   = boolopt "row-total" rawopts
-          ,no_total_    = boolopt "no-total" rawopts
-          ,show_costs_  = boolopt "show-costs" rawopts
-          ,sort_amount_ = boolopt "sort-amount" rawopts
-          ,percent_     = boolopt "percent" rawopts
-          ,invert_      = boolopt "invert" rawopts
-          ,pretty_tables_ = boolopt "pretty-tables" rawopts
-          ,color_       = useColorOnStdout -- a lower-level helper
-          ,forecast_    = forecastPeriodFromRawOpts d rawopts
-          ,transpose_   = boolopt "transpose" rawopts
-          }
+rawOptsToReportOpts :: Day -> RawOpts -> ReportOpts
```

Comment:
I should have put a brief haddock here, let's add one now mentioning that this raises an error if there are problems with the options (unparseable --format, --valuechange combined with non-end valuation, etc.)

Reply:
And if you notice any other error-raising functions without a haddock, let's mention it for those too: forecastPeriodFromRawOpts, rawOptsToInputOpts, ...